### PR TITLE
Allow multiple MAC addresses for 'on_ble_advertise' filter

### PR DESCRIPTION
**Pull request by RoboMagus**
# What does this implement/fix?

The `on_ble_advertise` automation trigger could be interesting for multiple MAC addresses, without disabling filtering altogether. Simply allowing users to specify a list solves that issue. In code we simply check if the MAC address in the trigger exists in the vector of all addresses to look for.

As the data provided to the automation ([ESPBtDevice](https://esphome.io/api/classesphome_1_1esp32__ble__tracker_1_1_e_s_p_b_t_device.html)) contains the MAC address anyway the distinction could still be made in the automation itself if required.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** fixes <link to issue>

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#2888

## Test Environment

- [x] ESP32
- [x] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:
```yaml
# Example config.yaml
esp32_ble_tracker:
  on_ble_advertise:
    - mac_address: 
        - 11:22:33:44:55:66
        - AA:BB:CC:DD:EE:FF
      then:
        - lambda: |-
            ESP_LOGD("ble_adv", "Found one of the devices of interest:");
            ESP_LOGD("ble_adv", "  address: %s", x.address_str().c_str());
            ESP_LOGD("ble_adv", "  name: %s", x.get_name().c_str());


```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

File: esphome/components/esp32_ble_tracker/automation.h [hunk 1]
```diff
@@ -10,18 +10,22 @@ namespace esp32_ble_tracker {
 class ESPBTAdvertiseTrigger : public Trigger<const ESPBTDevice &>, public ESPBTDeviceListener {
  public:
   explicit ESPBTAdvertiseTrigger(ESP32BLETracker *parent) { parent->register_listener(this); }
-  void set_address(uint64_t address) { this->address_ = address; }
+  void set_address(const std::vector<uint64_t> &addresses) { this->address_vec_ = addresses; }
```
Comment (jesserockz) [inline esphome/components/esp32_ble_tracker/automation.h]: Might as well rename the setter too...
```suggestion
  void set_addresses(const std::vector<uint64_t> &addresses) { this->address_vec_ = addresses; }
```

**Review by jesserockz — CHANGES_REQUESTED** [review]
Looks good, just one change (technically two) below.

File: esphome/components/esp32_ble_tracker/automation.h (hunk 1 above)
Comment (RoboMagus) [reply esphome/components/esp32_ble_tracker/automation.h]: Indeed I should have 👍 
It's fixed now.

**Comment by nielsnl68** [comment]
cool idea

**Review by jesserockz — APPROVED** [review]

**Pull request merged**